### ISSUE TITLE
Bad Request(400) is returned when calling Invoke-WebRequest with fragment(#)

### ISSUE THREAD

**Issue by yukihane** [PowerShell/PowerShell#7796]
Steps to reproduce
------------------

```powershell
Invoke-WebRequest http://localhost/hello#fragment
```

Expected behavior
-----------------

On PowerShell 5.1,

```powershell
Invoke-WebRequest http://localhost/hello#fragment
```

apache httpd access.log:

     [16/Sep/2018:10:52:18 +0000] "GET /hello HTTP/1.1" 404 203 "-" "Mozilla/5.0 (Windows NT; Windows NT 10.0; ja-JP) WindowsPowerShell/5.1.17134.228"

Actual behavior
---------------

apache httpd access.log:

    [16/Sep/2018:10:50:50 +0000] "GET /hello#fragment HTTP/1.1" 400 226 "-" "-"

The fragment is not cut off.

Environment data
----------------

```powershell
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.1.0
PSEdition                      Core
GitCommitId                    6.1.0
OS                             Linux 3.10.0-862.2.3.el7.x86_64 #1 SMP Wed May 9 18:05:47 UTC 2018
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by chuanjiao10** [comment]
#7368
docker module work fine with 6.04
docker module work fine with 6.1 preview2
docker module work **failure** with 6.1 preview3
docker module work **failure** with 6.1 ga
docker module rerurn 400 too，I don't know if it is the same problem, please help me to see:
![jing7796-001](https://user-images.githubusercontent.com/8968667/45612845-e4b92600-ba96-11e8-8cf2-444fdbbd3c8c.png)

**Comment by TravisEz13** [comment]
@chuanjiao10 the docker API's don't use fragments.  Your comment is unrelated to this issue.

**Comment by SteveL-MSFT** [comment]
Looks like a regression in corefx introduced by a fix they made

**Comment by SteveL-MSFT** [comment]
CoreFx has merged the change, we'll pick it up in the future once it shows up in a .Net Core release

**Comment by iSazonov** [comment]
The fix is 3.0 milestone - we have to wait months.

**Comment by CarloToso** [comment]
It seems like it is no longer an issue in PowerShell 7.3.
